### ISSUE TITLE
Invalid URL Error? 

### ISSUE THREAD

**Issue by DavidAwad** [watson-developer-cloud/personality-insights-python#1]
```bash
ERROR: Invalid URL u'<url>api/v2/profile': No schema supplied. Perhaps you meant http://<url>api/v2/profile?
127.0.0.1 - - [23/Mar/2015:12:27:54] "POST / HTTP/1.1" 200 101 "http://localhost:3000/" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_10_2) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/43.0.2329.0 Safari/537.36"
```

The online example linked in the readme works fine, running the example on my local machine has the logo flash and that's it unfortunately! 

It might be a requirements problem as the requirements.txt only contains three package names. 

Sorry for the errors lately, I'm looking to use this project for something and it would be AMAZING if this worked just like the Nodejs personality insights version. Thank you again.

**Comment by DavidAwad** [comment]
Wait, This is because there are no API credentials isn't it? 

I apologize.

**Comment by germanattanasio** [comment]
yeap! don't worry! make sure you switch to use Personality Insights because User Modeling will be removed tomorrow.

More about this here: http://www.ibm.com/smarterplanet/us/en/ibmwatson/developercloud/personality-insights.html

**Comment by DavidAwad** [comment]
So will this app be removed? Or is it using personality insights? It looks like it's using insights already. 

Also one other question since you're here. for running this locally, 

what does the credential object look like?  Where could I put my bluemix credentials? The reason being I want to hard code the credentials to get the app working first and then host it later. 

```python
        if vcapServices is not None:
            print("Parsing VCAP_SERVICES")
            services = json.loads(vcapServices)
            svcName = "user_modeling"
            if svcName in services:
                print("User modeling service found!")
                svc = services[svcName][0]["credentials"]
                self.url = svc["url"]
                self.username = svc["username"]
                self.password = svc["password"]
            else:
                print("ERROR: The User Modeling service was not found")
```

#THANK YOU

**Comment by DavidAwad** [comment]
This also Might be a stupid question given this code a few lines up. . .
```python
        # Local variables
        # Right here. Duh.
        self.url = "<url>"
        self.username = "<username>"
        self.password = "<password>"

        if vcapServices is not None:
            print("Parsing VCAP_SERVICES")
            services = json.loads(vcapServices)
            svcName = "user_modeling"
            if svcName in services:
                print("User modeling service found!")
                svc = services[svcName][0]["credentials"]
                self.url = svc["url"]
                self.username = svc["username"]
                self.password = svc["password"]
            else:
                print("ERROR: The User Modeling service was not found")
```

So do I need to edit this because user modeling no longer exists? Is that the case?

**Comment by germanattanasio** [comment]
You need to provide `url`, `username` and `password` if you want to run the app locally.
If you want to run it in Bluemix you need to change `svcName` to be `personality_insights`.

I recommend you to read the url from my previous answer which explains the Personality Insight service(previously known as User Modeling)

**Comment by DavidAwad** [comment]
okay, got it.
I am grabbing credentials from a different app I've got on bluemix using the personality insights service. 

I've hard coded the credentials and changed the `svcName` to `personality_insights`. 
I get an error saying `Internal server error (root cause: application/json)`. 
Is that referring to the headers in the request? 
```python 
    def getProfile(self, text):
        """Returns the profile by doing a POST to /v2/profile with text"""

        if self.url is None:
            raise Exception("No User Modeling service is bound to this app")

        data = self._formatPOSTData(text)
        response = requests.post(self.url + "/v2",
                          auth=(self.username, self.password),
                          headers = {"content-type": "application/json"},
                          data=json.dumps(data)
                          )
        try:
            return json.loads(response.text)
        except:
            raise Exception("Error processing the request, HTTP: %d" % response.status_code)

```

**Comment by germanattanasio** [comment]
> I am grabbing credentials from a different app 

You can't do that, you need to use Personality Insights credentials. If you use credentials from another service like Machine Translation or Relationship Extraction it will not work.

**Comment by DavidAwad** [comment]
oh no I meant another app using personality insights!

**Comment by germanattanasio** [comment]
@DavidAwad, I fixed #2  so this sample is now using Personality insights, you should be able to clone it and replace `url`, `username` and `password` to make it work locally.

**Comment by DavidAwad** [comment]
Works beautifully, thank you for all your help! 

I seriously can't describe to you how useful this will be.
